### ISSUE TITLE
static routes are always forced to replace

### ISSUE THREAD

**Issue by mobig** [netascode/terraform-provider-nxos#139]
when running a terraform job, the static routes are always a forced replacement. here is my resouse and local blocks

main.tf
```
  resource "nxos_ipv4_static_route" "static_routes" {
    for_each  = {for route in local.static_routes : route.key => route}
    device    = each.value.device
    vrf_name  = each.value.vrf
    prefix    = each.value.destination
    next_hops = [
      {
        interface_id = each.value.next_hop_interface == "" ? "unspecified" : each.value.next_hop_interface
        address      = each.value.next_hop_address
        vrf_name     = each.value.vrf
      }
    ]
  
    depends_on = [
      module.features,
      nxos_ipv4_vrf.default_vrf,
      nxos_vrf.vrf_config
    ]
  }
```

local.tf
```
  static_route_config_files = fileset(local.configs_path, "**/static_routes.csv")

  static_routes = flatten([
    for static_routes_config in local.static_route_config_files : [
      for static_route in csvdecode(file("${local.configs_path}${static_routes_config}")) :
      {
        key                = "${basename(dirname("${local.configs_path}${static_routes_config}"))}-${static_route.destination_network}"
        device             = basename(dirname("${local.configs_path}${static_routes_config}"))
        destination        = static_route.destination_network
        next_hop_address   = static_route.next_hop_ip
        next_hop_interface = static_route.next_hop_interface
        vrf                = static_route.vrf
        description        = static_route.description
        track_object       = static_route.track_object
        preference         = static_route.preference

      }
    ]
  ])
```

terraform apply output
```
# nxos_ipv4_static_route.static_routes["switch_2-9.9.9.0"] must be replaced
-/+ resource "nxos_ipv4_static_route" "static_routes" {
      ~ id        = "sys/ipv4/inst/dom-[default]/rt-[9.9.9.0]" -> (known after apply)
      ~ next_hops = [
          ~ {
              ~ interface_id = "" -> "Null0" # forces replacement
              ~ vrf_name     = "" -> "default" # forces replacement
                # (1 unchanged attribute hidden)
            },
        ]
        # (3 unchanged attributes hidden)
    }
```

i've checked this on the device and it is in fact removing the route and re-adding it. this will cause an outage for any routes that are redistributed

**Comment by mobig** [comment]
Here is a single file terraform script to replicate the issue. it's worth noting that when i do a resource per-device (not doing a for_each loop), i don't see the issue.
```
terraform {
  required_providers {
    nxos = {
      source  = "netascode/nxos"
      version = "0.4.0"
    }
  }
}

provider "nxos" {
  username = "<USER>"
  password = "<PW>"
  devices  = local.switches
}

locals {
  switches = [
    {
      name         = "switch_1"
      url          = "https://192.168.0.154"
      vrf          = "default"
      prefix       = "11.11.11.0/24"
      interface_id = "unspecified"
      address      = "5.2.2.2"
    },
    {
      name         = "switch_2"
      url          = "https://192.168.0.155"
      vrf          = "default"
      prefix       = "11.12.11.0/24"
      interface_id = "unspecified"
      address      = "5.2.2.1"
    }
  ]
}


resource "nxos_ipv4_static_route" "example" {
  for_each  = {for device in local.switches : device.name => device}
  device    = each.value.name
  vrf_name  = each.value.vrf
  prefix    = each.value.prefix
  next_hops = [
    {
      interface_id = each.value.interface_id
      address      = each.value.address
      vrf_name     = each.value.vrf
      description  = "My Description"
    }
  ]
}
```

**Comment by danischm** [comment]
Thanks for config that helped me recreate the issue. The issue should be fixed in v0.4.1 .
